internal/models/factor.go

```diff
@@ -236,6 +267,20 @@ func (f *Factor) IsVerified() bool {
 	return f.Status == FactorStateVerified.String()
 }
 
+func (f *Factor) FindChallengeByID(conn *storage.Connection, challengeID uuid.UUID) (*Challenge, error) {
+	var challenge Challenge
+	err := conn.Q().
+		Where("id = ?", challengeID).
+		Where("factor_id = ?", f.ID).
+		First(&challenge)
```

Comment:
You can collapse the `Where` to one line like:

```suggestion
	err := conn.Q().Where("id = ? and factor_id", challengeID, f.ID).First(&challenge)
```